cost-model/src/cost_model.rs

```diff
@@ -43,28 +45,13 @@ impl CostModel {
         transaction: &'a Tx,
         feature_set: &FeatureSet,
     ) -> TransactionCost<'a, Tx> {
-        if transaction.is_simple_vote_transaction() {
-            TransactionCost::SimpleVote { transaction }
-        } else {
-            let signature_cost = Self::get_signature_cost(transaction, feature_set);
-            let write_lock_cost = Self::get_write_lock_cost(transaction, feature_set);
-            let (programs_execution_cost, loaded_accounts_data_size_cost, data_bytes_cost) =
-                Self::get_transaction_cost(transaction, feature_set);
-            let allocated_accounts_data_size =
-                Self::calculate_allocated_accounts_data_size(transaction);
-
-            let usage_cost_details = UsageCostDetails {
-                transaction,
-                signature_cost,
-                write_lock_cost,
-                data_bytes_cost,
-                programs_execution_cost,
-                loaded_accounts_data_size_cost,
-                allocated_accounts_data_size,
-            };
-
-            TransactionCost::Transaction(usage_cost_details)
-        }
+        let num_write_locks = Self::num_write_locks(transaction, feature_set);
```

Comment:
Don't have to gather write locks if it's simple-vote. Could make all three public functions (calculate_cost, calculate_cost_for_executed_transaction, estimate_cost) have same 
```
if transaction.is_simple_vote_transaction() {
    return TransactionCost::SimpleVote { transaction };
}
// gather necessry stuff: num_write_locks, get_transaction_cost etc
Self::calculate_non_vote_transaction_cost(...)
}
```